### PR TITLE
Feature 68

### PR DESCRIPTION
**Pull request by klein2ms**
Implemented logging functionality for htmlviewer.  Accounts for css bug in htmlviewer and includes some refactoring for readability.

### CODE REVIEW

**Comment by jbhilton11** [comment]
Assuming that since this is saying there are no conflicts that the master as of yesterday afternoon has been re-merged into this branch and there are no issues.

File: src/htmlViewer/html-viewer.js [hunk 1]
```diff
@@ -1,89 +1,135 @@
...
+      let traceDataContainer = traceService.traceModel.traceDataContainer;
+      
+      let doc = this.getContentDocument();
+      let script = doc.createElement('script');
+      script.textContent = this.js;
+        
+      let result = undefined;
+      
+      try {
+        ea.publish(traceService.executionEvents.running.event);
```
Comment (jbhilton11) [inline src/htmlViewer/html-viewer.js]: Does publishing the 'running' and 'finished' events for the traceService in the addJs() method make sense? If I'm following the code correctly, the subscription to 'onJsEditorChanged' event in the subscribe() method above (line 34) starts the traceService execution, does it not? Would it make sense to move at least the publishing of the 'running' event in the subscribe() method right before line 34?

Comment (luminaxster) [inline src/htmlViewer/html-viewer.js]: In line 34 we get the text of the new code. It executes when added to the element  in line 61, [doc.body.appendChild(script);](https://github.com/tlatoza/SeeCodeRun/blob/feature-68/src/htmlViewer/html-viewer.js#L61)

Comment (jbhilton11) [inline src/htmlViewer/html-viewer.js]: Gotcha. Alright, when I get home I'll go ahead and merge this into master. Thanks for responding!

**Pull request merged**